_posts/en/newsletters/2019-01-15-newsletter.md

```diff
@@ -0,0 +1,94 @@
+---
+title: 'Bitcoin Optech Newsletter #29'
+permalink: /en/newsletters/2019/01/15/
+name: 2019-01-15-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter announces a security upgrade for C-Lightning,
+describes a paper and additional research into wallets that accidentally
+revealed their private keys, and lists some notable code changes in
+popular Bitcoin infrastructure projects.
+
+## Action items
+
+- **Upgrade to C-Lightning 0.6.3:** this release fixes a remote DoS
+  vulnerability that could be used to crash C-Lightning nodes and
+  potentially steal money.  See the *notable code changes* section below
+  for details.  This release also includes other less critical bug fixes
+  and new features.
+
+## News
+
+- **Weak signature nonces discovered:** a preprint [paper][weak nonces]
```

Comment:
May be worth adding somewhere that not reusing addresses avoids this issue entirely.

Reply:
Eh, what if you use RBF?  Furthermore, the Bitcore weak nonces only contained 64 bits of entropy, and the paper has this to say about that:

> 64-bit nonces and single-signature keys. Our lattice attack only applies when at least two signatures with a small nonce are created using the same secret key. A single 64-bit nonce requires only 2^32 time to break using Shanks’s baby-step-giant-step algorithm [34] or the Pollard rho algorithm [28], feasible with only modest computation resources. Applying this attack to all of the 2^30 Bitcoin signatures was beyond our resources, but we investigated a random sample.

I'm unfamiliar with the methods they describe, but IIUC it seems plausible that an attacker could scan their mempool for signatures with <=64-bit nonces and recover the corresponding private keys even with only one signature.

That said, if I ever start a band, I'm naming it Shanks's Baby-Step-Giant-Step Algorithm.

Reply:
Ah, you're right. Thanks for clarifying! Can I update my suggestion to "May be worth adding somewhere that not reusing addresses does not avoid this issue"? :slightly_smiling_face: 

Something like:

The method in the Breitner-Heninger paper was used to retrieve private keys where the address had been reused and the the key had been used to sign multiple transactions. However, even keys for addresses that have not been reused are vulnerable to attack if the nonce generation is biased or too small, either through using the same method if the key has been used to sign multiple times (eg for Replace-by-fee), or through simply brute-forcing the key using the baby-step giant-step or Pollard's Rho methods.